apps/heft/src/plugins/JestPlugin/jestWorkerPatch.ts

```diff
@@ -0,0 +1,132 @@
+import { Import, FileSystem } from '@rushstack/node-core-library';
+
+// This patch is a fix for a problem where Jest reports this error spuriously on a machine that is under heavy load:
+//
+// "A worker process has failed to exit gracefully and has been force exited. This is likely caused by tests
+// leaking due to improper teardown. Try running with --runInBand --detectOpenHandles to find leaks."
+//
+// The upstream issue is here: https://github.com/facebook/jest/issues/11354
+//
+// The relevant code is in jest-worker/src/base/BaseWorkerPool.ts:
+// https://github.com/facebook/jest/blob/64d5983d20a628d68644a3a4cd0f510dc304805a/packages/jest-worker/src/base/BaseWorkerPool.ts#L110
+//
+//      // Schedule a force exit in case worker fails to exit gracefully so
+//      // await worker.waitForExit() never takes longer than FORCE_EXIT_DELAY
+//      let forceExited = false;
+//      const forceExitTimeout = setTimeout(() => {
+//        worker.forceExit();
+//        forceExited = true;
+//      }, FORCE_EXIT_DELAY);
+//
+// The problem is that Jest hardwires FORCE_EXIT_DELAY to be 500 ms.  On a machine that is under heavy load,
+// the IPC message is not received from the child process before the timeout elapses.  The mitigation is to
+// increase the delay.  (Jest itself seems to be a significant contributor to machine load, so perhaps reducing
+// Jest's parallelism could also help.)
+
+interface IBaseWorkerPoolModule {
+  default: unknown;
+}
+
+// Follow the NPM dependency chain to find the module path for BaseWorkerPool.js
+// heft --> @jest/core --> @jest/reporters --> jest-worker
+
+const PATCHED_FORCE_EXIT_DELAY: number = 7000; // milliseconds
+
+function applyPatch(): void {
+  try {
+    let contextFolder: string = __dirname;
+    // Resolve the "@jest/core" package relative to Heft
+    contextFolder = Import.resolvePackage({ packageName: '@jest/core', baseFolderPath: contextFolder });
+    // Resolve the "@jest/reporters" package relative to "@jest/core"
+    contextFolder = Import.resolvePackage({ packageName: '@jest/reporters', baseFolderPath: contextFolder });
+    // Resolve the "jest-worker" package relative to "@jest/reporters"
+    const jestWorkerFolder: string = Import.resolvePackage({
+      packageName: 'jest-worker',
+      baseFolderPath: contextFolder
+    });
+
+    const baseWorkerPoolPath: string = path.join(jestWorkerFolder, 'build/base/BaseWorkerPool.js');
+    const baseWorkerPoolFilename: string = path.basename(baseWorkerPoolPath); // BaseWorkerPool.js
+
+    if (!FileSystem.exists(baseWorkerPoolPath)) {
+      throw new Error(
+        'The BaseWorkerPool.js file was not found in the expected location:\n' + baseWorkerPoolPath
+      );
+    }
+
+    // Load the module
+    const baseWorkerPoolModule: IBaseWorkerPoolModule = require(baseWorkerPoolPath);
+
+    // Obtain the metadata for the module
+    const baseWorkerPoolModuleMetadata: NodeModule | undefined = module.children.filter(
```

Comment:
Hmm is there a `findFirst`? Maybe just use a normal `for` loop and break when you find the expected thing?

Reply:
Done